burn-import/src/onnx/to_burn.rs

```diff
@@ -52,33 +51,81 @@ impl ModelCodeGen {
-        self.run();
+        self.run(true);
     }
 
     /// Run code generation.
-    pub fn run(&self) {
+    ///
+    /// This function is intended to be called from CLI.
+    pub fn run_from_cli(&self) {
+        self.run(false);
+    }
+
+    pub fn code_formatter() -> RustFmt {
         let config = Config::new_str()
             .post_proc(PostProcess::ReplaceMarkersAndDocBlocks)
             .edition(Edition::Rust2021);
 
-        let rust_formatter = RustFmt::from_config(config);
+        RustFmt::from_config(config)
+    }
+
+    /// Run code generation.
+    fn run(&self, is_build_script: bool) {
+        // prepend the out_dir to the cargo_out_dir if this is a build script
+        let out_dir = if is_build_script {
+            let cargo_out_dir = env::var("OUT_DIR").expect("OUT_DIR env is not set");
+            let mut path = PathBuf::from(cargo_out_dir);
+
+            // // Append the out_dir to the cargo_out_dir
+            path.push(self.out_dir.clone().unwrap());
+            path
+        } else {
+            self.out_dir.as_ref().expect("out_dir is not set").clone()
+        };
 
-        let out_dir = self.out_dir.as_ref().expect("out_dir is not set");
-        create_dir_all(out_dir).unwrap();
+        let rust_formatter = Self::code_formatter();
+
+        create_dir_all(&out_dir).unwrap();
 
         for input in self.inputs.iter() {
             let file_name = input.file_stem().unwrap();
-            let out_file = out_dir.join(file_name);
-            let out_file = out_file.with_extension("rs");
+            let out_file: PathBuf = out_dir.join(file_name);
 
-            let model = ModelSourceCode::new(input);
-            let code_str = rust_formatter.format_tokens(model.body()).unwrap();
+            if self.development {
+                let graph = parse_onnx(input.as_ref());
+                // export the graph
+                let debug_graph = format!("{:#?}", graph);
+                fs::write(out_file.with_extension("graph.txt"), debug_graph).unwrap();
+            }
 
-            fs::write(out_file, code_str).unwrap();
+            // export the source code
+            let model = ModelSourceCode::new(input, &out_file);
+            let code_str = rust_formatter.format_tokens(model.body()).unwrap();
+            fs::write(out_file.with_extension("rs"), code_str).unwrap();
+
+            // export the model state
+            if self.development {
+                // export the model state in debug mode
+                let model_state = ModelState::<DebugRecordSettings>::new_from_graph(model.graph);
+                model_state
+                    .record::<DebugRecordSettings>(out_file.clone())
+                    .unwrap()
+            } else {
+                let model_state = ModelState::<DefaultRecordSettings>::new_from_graph(model.graph);
+                model_state
+                    .record::<DefaultRecordSettings>(out_file)
+                    .unwrap();
+            }
```

Comment:
This should be extracted into a function (maybe) called `generate_model_state(input)`

Reply:
OK.